### PR TITLE
Test subscribe to an opstate cache path

### PR DESCRIPTION
**Pull request by ray-milkey**
This PR introduces two new subscribe tests to check on opstate cache paths that the device is updating. One used the diags API, one uses a straight GNMI call. This also refactors the existing subscribe tests to use utility routines to cut down on duplicate code.

### CODE REVIEW

File: test/gnmi/subscribetest.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019-present Open Networking Foundation.
+// Copyright 2020-present Open Networking Foundation.
```
Comment (SeanCondon) [inline test/gnmi/subscribetest.go]: probably shouldn't update if the file existed in 2019

Comment (ray-milkey) [reply test/gnmi/subscribetest.go]: Good catch. Not sure why I did that...
